src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.AsyncEnumerable.cs

```diff
@@ -0,0 +1,215 @@
+
+        [RequiresUnreferencedCode(HttpContentJsonExtensions.SerializationUnreferencedCodeMessage)]
+        [RequiresDynamicCode(HttpContentJsonExtensions.SerializationDynamicCodeMessage)]
+        private static IAsyncEnumerable<TValue?> FromJsonStreamAsyncCore<TValue>(
+            Func<HttpClient, Uri?, CancellationToken, Task<HttpResponseMessage>> getMethod,
+            HttpClient client,
+            Uri? requestUri,
+            JsonSerializerOptions? options,
+            CancellationToken cancellationToken)
+        {
+            options ??= JsonSerializerOptions.Default;
+            var jsonTypeInfo = (JsonTypeInfo<TValue>)options.GetTypeInfo(typeof(TValue));
+
+            return FromJsonStreamAsyncCore(getMethod, client, requestUri, jsonTypeInfo, cancellationToken);
+        }
+
+        private static IAsyncEnumerable<TValue?> FromJsonStreamAsyncCore<TValue>(
+            Func<HttpClient, Uri?, CancellationToken, Task<HttpResponseMessage>> getMethod,
+            HttpClient client,
+            Uri? requestUri,
+            JsonTypeInfo<TValue> jsonTypeInfo,
+            CancellationToken cancellationToken)
+        {
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            return Core(getMethod, client, requestUri, jsonTypeInfo, cancellationToken);
+
+            static async IAsyncEnumerable<TValue?> Core(
+                Func<HttpClient, Uri?, CancellationToken, Task<HttpResponseMessage>> getMethod,
+                HttpClient client,
+                Uri? requestUri,
+                JsonTypeInfo<TValue> jsonTypeInfo,
+                [EnumeratorCancellation] CancellationToken cancellationToken)
+            {
+                TimeSpan timeout = client.Timeout;
+
+                // Create the CTS before the initial SendAsync so that the SendAsync counts against the timeout.
+                CancellationTokenSource? linkedCTS = null;
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    linkedCTS = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                    linkedCTS.CancelAfter(timeout);
+                }
+
+                // We call SendAsync outside of the async Core method to propagate exception even without awaiting the returned task.
+                Task<HttpResponseMessage> responseTask;
+                try
+                {
+                    // Intentionally using cancellationToken instead of the linked one here as HttpClient will enforce the Timeout on its own for this part
+                    responseTask = getMethod(client, requestUri, cancellationToken);
+                }
+                catch
+                {
+                    linkedCTS?.Dispose();
+                    throw;
+                }
+
+                try
+                {
+                    using HttpResponseMessage response = await responseTask.ConfigureAwait(false);
```

Comment:
```suggestion
                using HttpResponseMessage response = await client.GetAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
```
And you can remove the finally block as well.

(See first half of https://github.com/dotnet/runtime/pull/89258#issuecomment-1645792740)